src/battle_util.c

```diff
@@ -10036,3 +10051,16 @@ bool32 IsBattlerWeatherAffected(u8 battlerId, u32 weatherFlags)
     }
     return FALSE;
 }
+
+// Gets move target before redirection effects etc. are applied
+// Possible return values are defined in battle.h following MOVE_TARGET_SELECTED
+u32 GetBattlerMoveTargetType(u8 battlerId, u16 move)
```

Comment:
Because of the `IsBattlerTerrainAffected` check, we probably should use a separate AI version of this with `AI_IsTerrainAffected` (because we effectively need to check `AI_IsBattlerGrounded`)